### PR TITLE
Add tablesort functionality to all tables

### PR DESCRIPTION
**Pull request by emmakemb**
### Short description
Short description

At the moment, only the event list can be sorted.
Proposed Solution

### Proposed changes

Add the same tablesort functionality to all our tables:

    Regions
    Languages
    Admin Users
    Region Users
    Organizations
    Offer Templates
    Admin Feedback
    Region Feedback
    Push Notifications

### Resolved issues

Fixes: #624

### CODE REVIEW

File: src/cms/templates/offer_templates/offer_template_list.html [hunk 1]
```diff
@@ -16,15 +16,15 @@ <h1 class="heading">{% trans 'Offer Templates' %}</h1>
...
-	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Slug' %}</th>
-	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'URL' %}</th>
-	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Last updated' %}</th>
-	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Created' %}</th>
-                <th class="text-sm text-left uppercase py-3 p2-4 pr-4 min">{% trans 'Options' %}</th>
+	            <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">{% trans 'Slug' %}</th>
+	            <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">{% trans 'URL' %}</th>
+	            <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">{% trans 'Last updated' %}</th>
+	            <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">{% trans 'Created' %}</th>
+                <th class="text-sm text-left uppercase py-3 p2-4 pr-4 minno-sort" data-sort-method="none">{% trans 'Options' %}</th>
```
Comment (timoludwig) [inline src/cms/templates/offer_templates/offer_template_list.html:27]:
```suggestion
	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Slug' %}</th>
	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'URL' %}</th>
	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Last updated' %}</th>
	            <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Created' %}</th>
                <th class="text-sm text-left uppercase py-3 p2-4 pr-4 min no-sort" data-sort-method="none">{% trans 'Options' %}</th>
```

File: src/cms/templates/organizations/organization_list.html [hunk 2]
```diff
@@ -16,13 +16,13 @@ <h1 class="heading">{% trans 'Manage Organizations' %}</h1>
...
         <thead>
             <tr class="border-b border-solid border-gray-200">
-                <th class="text-sm text-left uppercase py-3 pl-4 pr-2">{% trans 'Name' %}</th>
-                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Slug' %}</th>
-                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Thumbnail' %}</th>
-                <th class="text-sm text-left uppercase py-3 p2-4 pr-4 min">{% trans 'Options' %}</th>
+                <th class="text-sm text-left uppercase py-3 pl-4 pr-2 " >{% trans 'Name' %}</th>
+                <th class="text-sm text-left uppercase py-3 px-2 no-sort " data-sort-method="none">{% trans 'Slug' %}</th>
+                <th class="text-sm text-left uppercase py-3 px-2 no-sort " data-sort-method="none">{% trans 'Thumbnail' %}</th>
+                <th class="text-sm text-left uppercase py-3 p2-4 pr-4 min no-sort " data-sort-method="none">{% trans 'Options' %}</th>
```
Comment (timoludwig) [inline src/cms/templates/organizations/organization_list.html:25]:
```suggestion
                <th class="text-sm text-left uppercase py-3 pl-4 pr-2" >{% trans 'Name' %}</th>
                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Slug' %}</th>
                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Thumbnail' %}</th>
                <th class="text-sm text-left uppercase py-3 p2-4 pr-4 min no-sort" data-sort-method="none">{% trans 'Options' %}</th>
```

File: src/cms/templates/push_notifications/push_notification_list.html [hunk 3]
```diff
@@ -22,16 +22,16 @@ <h1 class="heading">{% trans 'News' %}</h1>
...
 <div class="table-listing">
-    <table class="w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white">
+    <table data-enable-table-sort class="sort w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white">
         <thead>
             <tr class="border-b border-solid border-gray-200">
                 <th class="text-sm text-left uppercase py-3 pl-4 pr-2">
                     {% trans 'Title' %}
                 </th>
-                <th class="text-sm text-left uppercase py-3 px-2">
+                <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">
```
Comment (timoludwig) [inline src/cms/templates/push_notifications/push_notification_list.html:31]:
```suggestion
                <th class="text-sm text-left uppercase py-3 px-2">
```

File: src/cms/templates/push_notifications/push_notification_list.html [hunk 4]
```diff
@@ -40,7 +40,7 @@ <h1 class="heading">{% trans 'News' %}</h1>
                         {% endfor %}
                     </div>
                 </th>
-                <th class="text-sm text-left uppercase py-3 px-2 min">
+                <th class="text-sm text-left uppercase py-3 px-2 min no-sort" data-sort-method="none">
```
Comment (timoludwig) [inline src/cms/templates/push_notifications/push_notification_list.html:43]:
```suggestion
                <th class="text-sm text-left uppercase py-3 px-2 min">
```

File: src/cms/templates/regions/region_list.html [hunk 5]
```diff
@@ -20,14 +20,14 @@ <h1 class="heading">{% trans 'Manage Regions' %}</h1>
...
         <thead>
             <tr class="border-b border-solid border-gray-200">
                 <th class="text-sm text-left uppercase py-3 pl-4 pr-2">{% trans 'Name' %}</th>
-                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Created' %}</th>
-                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Last updated' %}</th>
-                <th class="text-sm text-left uppercase py-3 px-2 min">{% trans 'Status' %}</th>
-                <th class="text-sm text-right uppercase py-3 pl-2 pr-4">{% trans 'Actions' %}</th>
+                <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">{% trans 'Created' %}</th>
+                <th class="text-sm text-left uppercase py-3 px-2 no-sort" data-sort-method="none">{% trans 'Last updated' %}</th>
+                <th class="text-sm text-left uppercase py-3 px-2 min no-sort" data-sort-method="none">{% trans 'Status' %}</th>
```
Comment (timoludwig) [inline src/cms/templates/regions/region_list.html:29]:
```suggestion
                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Created' %}</th>
                <th class="text-sm text-left uppercase py-3 px-2">{% trans 'Last updated' %}</th>
                <th class="text-sm text-left uppercase py-3 px-2 min">{% trans 'Status' %}</th>
```

File: src/cms/templates/users/admin/list.html [hunk 6]
```diff
@@ -20,18 +20,18 @@ <h1 class="heading">{% trans 'User Management' %}</h1>
...
-                <th class="text-sm text-left uppercase py-3 pl-2 pr-4 min">{% trans 'Administrator' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'First Name' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Last Name' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'E-mail-address' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Organization' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Last Login' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 min no-sort" data-sort-method="none">{% trans 'Active' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 min no-sort" data-sort-method="none">{% trans 'Staff' %}</th>
+                <th class="text-sm text-left uppercase py-3 pl-2 pr-4 min no-sort" data-sort-method="none">{% trans 'Administrator' %}</th>
```
Comment (timoludwig) [inline src/cms/templates/users/admin/list.html:34]:
```suggestion
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'First Name' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Last Name' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'E-mail-address' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Organization' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Last Login' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Active' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2 min">{% trans 'Staff' %}</th>
                <th class="text-sm text-left uppercase py-3 pl-2 pr-4 min">{% trans 'Administrator' %}</th>
```

File: src/cms/templates/users/region/list.html [hunk 7]
```diff
@@ -20,16 +20,16 @@ <h1 class="heading">{% trans 'User Management' %}</h1>
...
-                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'E-mail-address' %}</th>
-                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Organization' %}</th>
-                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Last Login' %}</th>
-                <th class="text-sm text-left uppercase py-3 pr-4 min">{% trans 'Active' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'First Name' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Last Name' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'E-mail-address' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Organization' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Last Login' %}</th>
+                <th class="text-sm text-left uppercase py-3 pr-4 min no-sort" data-sort-method="none">{% trans 'Active' %}</th>
```
Comment (timoludwig) [inline src/cms/templates/users/region/list.html:32]:
```suggestion
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'First Name' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Last Name' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'E-mail-address' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Organization' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-2">{% trans 'Last Login' %}</th>
                <th class="text-sm text-left uppercase py-3 pr-4 min">{% trans 'Active' %}</th>
```

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks for your PR! :+1:

I think there was a misunderstanding: `data-sort-method="none"` should only be added when the column does not contain data which can be sorted, such as:
- Checkboxes
- Country Flags
- Options/Actions

In all other cases, we want to keep the option of sorting by that column (not just the first column).

File: src/cms/templates/feedback/admin_feedback_list.html [hunk 8]
```diff
@@ -24,18 +24,18 @@ <h1 class="heading">{% trans 'Technical Feedback' %}</h1>
...
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Category' %}</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Feedback on' %}</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Language' %}</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Comment' %}</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Rating' %}</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Rating' %} ({% trans 'overall' %})</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Region' %}</th>
+            <th class="text-sm text-left uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Marked as read by' %}</th>
+            <th class="text-sm text-right uppercase py-3 pr-2 no-sort" data-sort-method="none">{% trans 'Date' %}</th>
```
[the remark this answers is not in the document]
Comment (mael-fosso) [reply src/cms/templates/feedback/admin_feedback_list.html:38]: for me it should not have made sense to perform a sort according to these columns

File: src/cms/templates/feedback/admin_feedback_list.html [hunk 9]
```diff
@@ -24,18 +24,18 @@ <h1 class="heading">{% trans 'Technical Feedback' %}</h1>
 
 <form id="bulk-action-form" method="POST">
     {% csrf_token %}
-    <table class="w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white" data-enable-row-checkbox-toggle>
+    <table data-enable-table-sort class="sort w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white" data-enable-row-checkbox-toggle>
         <tr class="border-b border-solid border-gray-200">
```
[the remark this answers is not in the document]
Comment (mael-fosso) [reply src/cms/templates/feedback/admin_feedback_list.html:28]: it escaped me

File: src/cms/templates/feedback/admin_feedback_list.html [hunk 10]
```diff
@@ -24,18 +24,18 @@ <h1 class="heading">{% trans 'Technical Feedback' %}</h1>
...
             <th class="text-sm text-left uppercase py-3 pr-2" style="padding-left: 10px;" title="{% trans 'Select all' %}"><input type="checkbox" id="bulk-select-all"></th>
```
[the remark this answers is not in the document]
Comment (mael-fosso) [reply src/cms/templates/feedback/admin_feedback_list.html]: ok

[the remark this answers is not in the document]
Comment (timoludwig) [reply src/cms/templates/feedback/admin_feedback_list.html]: This is not resolved yet, please add `data-sort-method="none"` and the class `no-sort` for the checkbox column of the admin feedback list.

File: src/cms/templates/feedback/region_feedback_list.html [hunk 11]
```diff
@@ -24,17 +24,17 @@ <h1 class="heading">{% trans 'Feedback' %}</h1>
 
 <form id="bulk-action-form" method="POST">
     {% csrf_token %}
-    <table class="w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white" data-enable-row-checkbox-toggle>
+    <table data-enable-table-sort class="sort w-full mt-4 rounded border border-solid border-gray-200 shadow bg-white" data-enable-row-checkbox-toggle>
         <tr class="border-b border-solid border-gray-200">
             <th class="text-sm text-left uppercase py-3 pr-2" style="padding-left: 10px;" title="{% trans 'Select all' %}"><input type="checkbox" id="bulk-select-all"></th>
```
[the remark this answers is not in the document]
Comment (timoludwig) [reply src/cms/templates/feedback/region_feedback_list.html]: This is not resolved yet either.

File: src/cms/templates/push_notifications/push_notification_list.html (hunk 3 above)
Comment (timoludwig) [reply src/cms/templates/push_notifications/push_notification_list.html]: This is not solved yet.

File: src/cms/templates/push_notifications/push_notification_list.html (hunk 4 above)
Comment (timoludwig) [reply src/cms/templates/push_notifications/push_notification_list.html:43]: This is not resolved yet.

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks for your changes! :+1:
Sorry that it took me so long to review your PR!
In general it looks good, there are just a few comments from last time which are not resolved yet. I commented each entry which is still missing.
Thanks again!

**Comment by ulliholtgrave** [comment]
As we are no longer needing this changes I'll close this PR for now, but thank you @mael-fosso & @emmakemb  very much for your effort :+1: :)

**Pull request closed without merging**